Issue/228 wide images dont fit lists

hypest opened the pull request:
### Fix #228 

Dynamically scales image when line width is constraint.

### Test
1. In the demo app, put the cursor on the left of the image already there (the Aztec 1600x600px image)
2. Tap the "list" button to start a list
3. Watch the image shrink in size to fit the line/width
4. Rotate the device to landscape. Image should expand to accommodate the extra width now available
5. Rotate back to portrait. Image should shrink again

### Review
@0nko want to pick this up? Thanks!

theck13 commented on aztec/src/main/kotlin/org/wordpress/aztec/AztecText.kt:
```diff
@@ -603,18 +615,15 @@ class AztecText : EditText, TextWatcher {
-             * https://material.io/guidelines/layout/metrics-keylines.html#metrics-keylines-baseline-grids
-             */
-            val width = context.resources.displayMetrics.widthPixels - DisplayUtils.dpToPx(context, 32)
+            // maxWidth set to the biggest of screen width/height to cater for device rotation
```
Is `maxWidth` a typo in the comment?  It seems like it should be `width`.

hypest replied:
Hey @theck13 , thanks for picking this up for review!

Actually no, it's not a typo. The parameter in `imageGetter?.loadImage` is actually about the maximum image wanted to be returned. I renamed the variable with b8bfe38 to better match that meaning. Thanks!

theck13 commented on app/src/main/kotlin/org/wordpress/aztec/demo/MainActivity.kt:
```diff
@@ -212,9 +221,11 @@ class MainActivity : AppCompatActivity(), OnRequestPermissionsResultCallback, Vi
+        if (savedInstanceState == null) {
+            aztec.fromHtml(source.getPureHtml())
+            source.history = aztec.history
+        }
```
This causes a crash if the history is not saved (i.e. `onSaveInstanceState`) and restored (i.e. `onRestoreInstanceState`) on device rotation.  Launch the app, rotate the device, tap the ***HTML*** format button, and the app crashes.  See the stack trace below.

```
02-14 09:41:46.935 7403-7403/org.wordpress.aztec E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.wordpress.aztec, PID: 7403
    kotlin.UninitializedPropertyAccessException: lateinit property history has not been initialized
        at org.wordpress.aztec.source.SourceViewEditText.beforeTextChanged(SourceViewEditText.kt:121)
        at android.widget.TextView.sendBeforeTextChanged(TextView.java:8137)
        at android.widget.TextView.setText(TextView.java:4380)
        at android.widget.TextView.setText(TextView.java:4337)
        at android.widget.EditText.setText(EditText.java:89)
        at android.widget.TextView.setText(TextView.java:4312)
        at org.wordpress.aztec.source.SourceViewEditText.displayStyledAndFormattedHtml(SourceViewEditText.kt:166)
        at org.wordpress.aztec.toolbar.AztecToolbar.toggleEditorMode(AztecToolbar.kt:179)
        at org.wordpress.aztec.demo.MainActivity.onToolbarHtmlModeClicked(MainActivity.kt:547)
        at org.wordpress.aztec.toolbar.AztecToolbar.onToolbarAction(AztecToolbar.kt:170)
        at org.wordpress.aztec.toolbar.AztecToolbar.access$onToolbarAction(AztecToolbar.kt:18)
        at org.wordpress.aztec.toolbar.AztecToolbar$initView$1.onClick(AztecToolbar.kt:96)
        at android.view.View.performClick(View.java:5637)
        at android.widget.CompoundButton.performClick(CompoundButton.java:122)
        at android.view.View$PerformClick.run(View.java:22429)
        at android.os.Handler.handleCallback(Handler.java:751)
        at android.os.Handler.dispatchMessage(Handler.java:95)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6119)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
```

hypest replied:
Nice catch :)

Fixed with 41e4661.

theck13 commented on aztec/src/main/kotlin/org/wordpress/aztec/spans/AztecMediaSpan.kt:
```diff
@@ -68,25 +64,112 @@ class AztecMediaSpan(val context: Context, private var drawable: Drawable?,
+            width = widthSize
+        } else {
+            if (des < 0) {
+                boring = BoringLayout.isBoring("", textView?.paint)
```
Ha, is this _really_ the name of a layout?!

hypest replied:
Hehe, it is, yeah, and apparently a boring one :)

hypest commented:
I've addressed your comments @theck13 so those are ready for another pass. But, I've seen another rotation crash (change to HTML and just rotate) so I will work on that now. You may want to postpone the next review pass until that's fixed as well. I'll let you know. Thanks!
